x/gamm/pool-models/stableswap/amm.go

```diff
@@ -441,9 +441,24 @@ func (p *Pool) calcSingleAssetJoinShares(tokenIn sdk.Coin, swapFee sdk.Dec) (sdk
 		return &paCopy
 	}
 
-	// We apply the swap fee by multiplying by (1 - swapFee) and then truncating to int
-	oneMinusSwapFee := sdk.OneDec().Sub(swapFee)
-	tokenInAmtAfterFee := tokenIn.Amount.ToDec().Mul(oneMinusSwapFee).TruncateInt()
+	// Find the portion of the pool made up by non-input assets
+	sumScalingFactors := int64(0)
```

Comment:
Consider encapsulating the new logic into a function with a godoc and adding a couple of basic tests

Reply:
Suggest elaborating in the spec on this point: `only charge swap fee on the portion of the input tokens made of non-input assets`. Also, answering why we need to charge the fee only on the proportion of non-input assets would be helpful.